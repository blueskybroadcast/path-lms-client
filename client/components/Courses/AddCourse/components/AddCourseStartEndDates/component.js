import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';

const AddCourseStartEndDates = ({
  startDate, endDate, handleStartDateChange, handleEndDateChange
}) => (
  <fieldset>
    <legend>
      <span>Start Date</span>
    </legend>
    <div className="control-group string optional course_start_date">
      <label
        className="string optional control-label"
        htmlFor="course_start_date"
      >
        Start Date
      </label>
      <div className="controls">
        <DatePicker
          selected={startDate}
          onChange={handleStartDateChange}
          name="startDate"
          id="course_start_date"
          className="course_end_date_datepicker picker__input"
        />
      </div>
    </div>
    <legend>
      <span>End Date</span>
    </legend>
    <div className="control-group string optional course_end_date">
      <label
        className="string optional control-label"
        htmlFor="course_end_date"
      >
        End Date
      </label>
      <div className="controls">
        <DatePicker
          selected={endDate}
          onChange={handleEndDateChange}
          name="endDate"
          id="course_end_date"
          className="course_end_date_datepicker picker__input"
        />
      </div>
    </div>
    <div className="control-group cover-photo-wrapper">
      <div className="controls hint-container">
        <p className="hint">Course start and end dates are visible to admins and included in reports</p>
      </div>
    </div>
  </fieldset>
);

AddCourseStartEndDates.propTypes = {
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
  handleStartDateChange: PropTypes.func.isRequired,
  handleEndDateChange: PropTypes.func.isRequired
};

AddCourseStartEndDates.defaultProps = {
  startDate: null,
  endDate: null
};

export default AddCourseStartEndDates;
