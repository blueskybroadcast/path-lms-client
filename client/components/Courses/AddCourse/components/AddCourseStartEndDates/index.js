import AddCourseStartEndDates from './component';

export default AddCourseStartEndDates;
