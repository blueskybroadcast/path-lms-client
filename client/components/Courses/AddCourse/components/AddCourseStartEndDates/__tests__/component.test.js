import React from 'react';
import { shallow } from 'enzyme';
import AddCourseStartEndDates from '../component';

describe('AddCourseStartEndDates', () => {
  const props = {
    handleStartDateChange: jest.fn(),
    handleEndDateChange: jest.fn()
  };

  test('should render AddCourseStartEndDates without dates correctly', () => {
    const wrapper = shallow(
      <AddCourseStartEndDates {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render AddCourseStartEndDates with dates correctly', () => {
    const wrapper = shallow(
      <AddCourseStartEndDates
        {...props}
        startDate={new Date(Date.UTC(2019, 3, 10))}
        endDate={new Date(Date.UTC(2019, 3, 20))}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
