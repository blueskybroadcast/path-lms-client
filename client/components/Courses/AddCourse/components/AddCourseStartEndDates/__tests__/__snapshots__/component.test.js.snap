// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseStartEndDates should render AddCourseStartEndDates with dates correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Start Date
    </span>
  </legend>
  <div
    className="control-group string optional course_start_date"
  >
    <label
      className="string optional control-label"
      htmlFor="course_start_date"
    >
      Start Date
    </label>
    <div
      className="controls"
    >
      <DatePicker
        allowSameDay={false}
        className="course_end_date_datepicker picker__input"
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        id="course_start_date"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        name="startDate"
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2019-04-10T00:00:00.000Z}
        shouldCloseOnSelect={true}
        showTimeSelect={false}
        timeCaption="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </div>
  </div>
  <legend>
    <span>
      End Date
    </span>
  </legend>
  <div
    className="control-group string optional course_end_date"
  >
    <label
      className="string optional control-label"
      htmlFor="course_end_date"
    >
      End Date
    </label>
    <div
      className="controls"
    >
      <DatePicker
        allowSameDay={false}
        className="course_end_date_datepicker picker__input"
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        id="course_end_date"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        name="endDate"
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2019-04-20T00:00:00.000Z}
        shouldCloseOnSelect={true}
        showTimeSelect={false}
        timeCaption="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </div>
  </div>
  <div
    className="control-group cover-photo-wrapper"
  >
    <div
      className="controls hint-container"
    >
      <p
        className="hint"
      >
        Course start and end dates are visible to admins and included in reports
      </p>
    </div>
  </div>
</fieldset>
`;

exports[`AddCourseStartEndDates should render AddCourseStartEndDates without dates correctly 1`] = `
<fieldset>
  <legend>
    <span>
      Start Date
    </span>
  </legend>
  <div
    className="control-group string optional course_start_date"
  >
    <label
      className="string optional control-label"
      htmlFor="course_start_date"
    >
      Start Date
    </label>
    <div
      className="controls"
    >
      <DatePicker
        allowSameDay={false}
        className="course_end_date_datepicker picker__input"
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        id="course_start_date"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        name="startDate"
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={null}
        shouldCloseOnSelect={true}
        showTimeSelect={false}
        timeCaption="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </div>
  </div>
  <legend>
    <span>
      End Date
    </span>
  </legend>
  <div
    className="control-group string optional course_end_date"
  >
    <label
      className="string optional control-label"
      htmlFor="course_end_date"
    >
      End Date
    </label>
    <div
      className="controls"
    >
      <DatePicker
        allowSameDay={false}
        className="course_end_date_datepicker picker__input"
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        id="course_end_date"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        name="endDate"
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[MockFunction]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={null}
        shouldCloseOnSelect={true}
        showTimeSelect={false}
        timeCaption="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </div>
  </div>
  <div
    className="control-group cover-photo-wrapper"
  >
    <div
      className="controls hint-container"
    >
      <p
        className="hint"
      >
        Course start and end dates are visible to admins and included in reports
      </p>
    </div>
  </div>
</fieldset>
`;
